src/main/scala/cromwell/engine/backend/jes/Run.scala

```diff
@@ -25,7 +25,10 @@ object Run  {
 
   def apply(pipeline: Pipeline): Run = {
     val rpr = new RunPipelineRequest().setPipelineId(pipeline.id).setProjectId(pipeline.projectId).setServiceAccount(JesServiceAccount)
-    val tag = s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}]"
+    def tag: String = {
+      val shard = pipeline.key.index.map(x => s":$x").getOrElse("")
+      s"JES Run [UUID(${pipeline.workflow.shortId}):${pipeline.key.scope.name}$shard]"
+    }
```

Comment:
Yay centralizing this logic!  Moving the `val shard...` line to the `IndexEnhancedIndex` implicit class 

```scala
override def toString = index map { ":" + _ } getOrElse ""
```
would make that more generally reusable and address @geoffjentry's complaint about the anon function here

(and maybe `CallKey` should have a nice `toString` that composes the scope name with the shard's `toString`)